roomdraw_backend.py

```diff
@@ -0,0 +1,175 @@
+    first_name: str
+    last_name: str
+    email: str
+    class_year: str
+    priority_number: int
+
+class DrawLogic:
+    """
+    Performs the logic functions of the HMC Room Draw Backend. 
+    Satisifies requirement 3 in the Digital Draw Application Requirements.
+    """
+    def __init__(self):
+        None 
+    
+    def comparePriority(self, student1:UserId, student2:UserId) -> User:
+        """Compares class year and priority numbers of student1 and student2.
+           Returns the student that has higher priority
+        """
+        # Retrieves the student1 and student2 object from the database
+        # Compares the class_year attribute of student 1 and student 2 
+        # (Error-checking for valid class year)
+        # Compares the priority_number attribute of student 1 and student 2
+        # (Error-checking for valid priority numbers)
+        # Returns the student that has higher priority
+        return False
+    
+    def comparePriority(self, student1:User, student2:User) -> User:
+        """Compares class year and priority numbers of student1 and student2.
+           Returns the student that has higher priority
+        """
+        # Compares the class_year attribute of student 1 and student 2
+        # (Error-checking for valid class year)
+        # Compares the priority_number attribute of student 1 and student 2
+        # (Error-checking for valid priority numbers)
+        # Returns the student that has higher priority
+        return None
+
+class DrawState:
+    """Returns the total state of what goes on in the digital draw.
+    Helps satisfiy requirement 7 in Digital Draw Application Requirements
+    """
+    def __init__(self):
+        None
+    
+    def getUserIn(self, room: RoomId) -> Optional[User]:
+        """getUserIn takes in a RoomId and finds the user at that specific room. 
+        Returns the student or None if there is no student.
+        """
+        # Retrieves the room object from the database by calling getRoom (room:RoomId) 
+        # (Error-checking for making sure the room object is valid) 
+        # Retrieves the user object from the database by calling getUserIn (room:Room) 
+        # (Error-checking for making sure the user object is valid) 
+        return None
+    
+    def getRoomOf(self, user: UserId) -> Optional[Room]:
+        """getRoomOf takes in a UserId and finds the room the user is currently living in. 
+        Returns the student or None if there is no student. 
+        """
+        # Retrieves the user object from the database by calling getUser (id:UserId) 
+        # (Error-checking for making sure the user object is valid)
+        # Retrieves the room object from the database by calling getRoomOf (user:User)
+        # # (Error-checking for making sure the room object is valid) 
+        return None
+
+class DrawAction:
+    """
+    Potential actions a user can perform (frequently interacts with DrawState)
+    """
+    def __init__(self):
```

Comment:
ISSUE: Add the repository as a parameter of `__init__`.

```suggestion
    def __init__(self, repository):
        self.repository = repository
```

Then, in the functions, you can access the repository like this `user = self.repository.getUser(1)`.

Reply:
Adi agrees that this is ok.